--- liteeth_mac_wb_mii.v.orig	2025-09-16 15:03:57.271685200 -0700
+++ liteeth_mac_wb_mii.v	2025-09-16 15:06:32.548724800 -0700
@@ -18,7 +18,7 @@
 // Module
 //------------------------------------------------------------------------------
 
-module liteeth_core (
+module liteeth_mac_wb_mii (
     output wire          interrupt,
     input  wire          mii_clocks_rx,
     input  wire          mii_clocks_tx,
@@ -3851,43 +3851,41 @@
 
 
 //------------------------------------------------------------------------------
-// Memory mac_sram_writer_slot0: 383-words x 32-bit
+// Modified Memory mac_sram_writer_slot0: 384-words x 32-bit
 //------------------------------------------------------------------------------
 // Port 0 | Read: Sync  | Write: Sync | Mode: Write-First
 // Port 1 | Read: Sync  | Write: ---- | 
-reg [31:0] mac_sram_writer_slot0[0:382];
-reg [8:0] mac_sram_writer_slot0_adr0;
-reg [31:0] mac_sram_writer_slot0_dat1;
-always @(posedge sys_clk) begin
-	if (wishbone_interface_writer_memory0_we)
-		mac_sram_writer_slot0[wishbone_interface_writer_memory0_adr] <= wishbone_interface_writer_memory0_dat_w;
-	mac_sram_writer_slot0_adr0 <= wishbone_interface_writer_memory0_adr;
-end
-always @(posedge sys_clk) begin
-	mac_sram_writer_slot0_dat1 <= mac_sram_writer_slot0[wishbone_interface_sram0_adr];
-end
-assign wishbone_interface_writer_memory0_dat_r = mac_sram_writer_slot0[mac_sram_writer_slot0_adr0];
-assign wishbone_interface_sram0_dat_r = mac_sram_writer_slot0_dat1;
+(* keep *) fakeram_1rw1r_32w384d_sram u_tx_buffer_0 (
+    .rw0_clk(sys_clk),
+    .rw0_ce_in(1'b1),
+    .rw0_we_in(wishbone_interface_writer_memory0_we),
+    .rw0_addr_in(wishbone_interface_writer_memory0_adr),
+    .rw0_wd_in(wishbone_interface_writer_memory0_dat_w),
+    .rw0_rd_out(wishbone_interface_writer_memory0_dat_r),    
+    .r0_clk(sys_clk),
+    .r0_ce_in(1'b1),
+    .r0_addr_in(wishbone_interface_sram0_adr),
+    .r0_rd_out(wishbone_interface_sram0_dat_r)
+);
 
 
 //------------------------------------------------------------------------------
-// Memory mac_sram_writer_slot1: 383-words x 32-bit
+// Modified Memory mac_sram_writer_slot1: 384-words x 32-bit
 //------------------------------------------------------------------------------
 // Port 0 | Read: Sync  | Write: Sync | Mode: Write-First
 // Port 1 | Read: Sync  | Write: ---- | 
-reg [31:0] mac_sram_writer_slot1[0:382];
-reg [8:0] mac_sram_writer_slot1_adr0;
-reg [31:0] mac_sram_writer_slot1_dat1;
-always @(posedge sys_clk) begin
-	if (wishbone_interface_writer_memory1_we)
-		mac_sram_writer_slot1[wishbone_interface_writer_memory1_adr] <= wishbone_interface_writer_memory1_dat_w;
-	mac_sram_writer_slot1_adr0 <= wishbone_interface_writer_memory1_adr;
-end
-always @(posedge sys_clk) begin
-	mac_sram_writer_slot1_dat1 <= mac_sram_writer_slot1[wishbone_interface_sram1_adr];
-end
-assign wishbone_interface_writer_memory1_dat_r = mac_sram_writer_slot1[mac_sram_writer_slot1_adr0];
-assign wishbone_interface_sram1_dat_r = mac_sram_writer_slot1_dat1;
+(* keep *) fakeram_1rw1r_32w384d_sram u_tx_buffer_1 (
+    .rw0_clk(sys_clk),
+    .rw0_ce_in(1'b1),
+    .rw0_we_in(wishbone_interface_writer_memory1_we),
+    .rw0_addr_in(wishbone_interface_writer_memory1_adr),
+    .rw0_wd_in(wishbone_interface_writer_memory1_dat_w),
+    .rw0_rd_out(wishbone_interface_writer_memory1_dat_r),    
+    .r0_clk(sys_clk),
+    .r0_ce_in(1'b1),
+    .r0_addr_in(wishbone_interface_sram1_adr),
+    .r0_rd_out(wishbone_interface_sram1_dat_r)
+);
 
 
 //------------------------------------------------------------------------------
@@ -3909,49 +3907,43 @@
 
 
 //------------------------------------------------------------------------------
-// Memory mac_sram_reader_slot0: 383-words x 32-bit
+// Modified Memory mac_sram_reader_slot0: 384-words x 32-bit
 //------------------------------------------------------------------------------
 // Port 0 | Read: Sync  | Write: ---- | 
 // Port 1 | Read: Sync  | Write: Sync | Mode: Write-First | Write-Granularity: 8
-reg [31:0] mac_sram_reader_slot0[0:382];
-reg [31:0] mac_sram_reader_slot0_dat0;
-reg [8:0] mac_sram_reader_slot0_adr1;
-always @(posedge sys_clk) begin
-	if (wishbone_interface_reader_memory0_re)
-		mac_sram_reader_slot0_dat0 <= mac_sram_reader_slot0[wishbone_interface_reader_memory0_adr];
-end
-always @(posedge sys_clk) begin : mem_write_block
-	integer we_index;
-	for(we_index = 0; we_index < 4; we_index=we_index+1)
-		if (wishbone_interface_sram2_we[we_index])
-			mac_sram_reader_slot0[wishbone_interface_sram2_adr][we_index*8 +: 8] <= wishbone_interface_sram2_dat_w[we_index*8 +: 8];
-	mac_sram_reader_slot0_adr1 <= wishbone_interface_sram2_adr;
-end
-assign wishbone_interface_reader_memory0_dat_r = mac_sram_reader_slot0_dat0;
-assign wishbone_interface_sram2_dat_r = mac_sram_reader_slot0[mac_sram_reader_slot0_adr1];
+(* keep *) fakeram_1rw1r_32w384d_8wm_sram u_rx_buffer_0 (
+    .rw0_clk(sys_clk),
+    .rw0_ce_in(1'b1),
+    .rw0_we_in(wishbone_interface_sram2_we),
+    .rw0_wmask_in(wishbone_interface_sram2_we),
+    .rw0_addr_in(wishbone_interface_sram2_adr),
+    .rw0_wd_in(wishbone_interface_sram2_dat_w),
+    .rw0_rd_out(wishbone_interface_sram2_dat_r),
+    .r0_clk(sys_clk),
+    .r0_ce_in(wishbone_interface_reader_memory0_re),
+    .r0_addr_in(wishbone_interface_reader_memory0_adr),
+    .r0_rd_out(wishbone_interface_reader_memory0_dat_r)
+);
 
 
 //------------------------------------------------------------------------------
-// Memory mac_sram_reader_slot1: 383-words x 32-bit
+// Modified Memory mac_sram_reader_slot1: 384-words x 32-bit
 //------------------------------------------------------------------------------
 // Port 0 | Read: Sync  | Write: ---- | 
 // Port 1 | Read: Sync  | Write: Sync | Mode: Write-First | Write-Granularity: 8
-reg [31:0] mac_sram_reader_slot1[0:382];
-reg [31:0] mac_sram_reader_slot1_dat0;
-reg [8:0] mac_sram_reader_slot1_adr1;
-always @(posedge sys_clk) begin
-	if (wishbone_interface_reader_memory1_re)
-		mac_sram_reader_slot1_dat0 <= mac_sram_reader_slot1[wishbone_interface_reader_memory1_adr];
-end
-always @(posedge sys_clk) begin : mem_write_block_1
-	integer we_index_1;
-	for(we_index_1 = 0; we_index_1 < 4; we_index_1=we_index_1+1)
-		if (wishbone_interface_sram3_we[we_index_1])
-			mac_sram_reader_slot1[wishbone_interface_sram3_adr][we_index_1*8 +: 8] <= wishbone_interface_sram3_dat_w[we_index_1*8 +: 8];
-	mac_sram_reader_slot1_adr1 <= wishbone_interface_sram3_adr;
-end
-assign wishbone_interface_reader_memory1_dat_r = mac_sram_reader_slot1_dat0;
-assign wishbone_interface_sram3_dat_r = mac_sram_reader_slot1[mac_sram_reader_slot1_adr1];
+(* keep *) fakeram_1rw1r_32w384d_8wm_sram u_rx_buffer_1 (
+    .rw0_clk(sys_clk),
+    .rw0_ce_in(1'b1),
+    .rw0_we_in(wishbone_interface_sram3_we),
+    .rw0_wmask_in(wishbone_interface_sram3_we),
+    .rw0_addr_in(wishbone_interface_sram3_adr),
+    .rw0_wd_in(wishbone_interface_sram3_dat_w),
+    .rw0_rd_out(wishbone_interface_sram3_dat_r),
+    .r0_clk(sys_clk),
+    .r0_ce_in(wishbone_interface_reader_memory1_re),
+    .r0_addr_in(wishbone_interface_reader_memory1_adr),
+    .r0_rd_out(wishbone_interface_reader_memory1_dat_r)
+);
 
 
 (* ars_ff1 = "true", async_reg = "true" *)
